Highway.h

```diff
@@ -2,9 +2,16 @@
 
 #include <vector>
 #include <algorithm>
+#include "Vehicle.h"
+#include "Car.h"
+#include "Motorcycle.h"
+#include "SemiTruck.h"
```

Comment:
Don't include what you don't use.
If you aren't using any member variables from these types IN THIS HEADER FILE, 
don't include the header files here.  Forward declare instead.
Include them in the cpp file.
- [ ] revise